roles/zookeeper/tasks/main.yml

```diff
@@ -81,4 +80,24 @@
 
 - include: collectd.yml
 
+- name: stop zookeeper service
+  sudo: yes
+  service:
+    name: "{{ zookeeper_service }}"
+    state: stopped
+  when: repair_mode is defined
+  tags:
+    - zookeeper
+
+- name: register zookeeper service
+  sudo: yes
+  service:
+    enabled: yes
+    name: "{{ zookeeper_service }}"
+    state: started
+  notify:
+    - wait for zookeeper to listen
+  tags:
+    - zookeeper
+
```

Comment:
We do start in any case, but stop only if repair=true.  We don't want to restart every time.